lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -413,8 +413,14 @@ void storeDistributedToShared(MemDescType dstTy, RankedTensorType srcTy,
                               const TargetInfoBase &target, bool crossGrain) {
   bool success;
   std::function<void(VectorType, Value /*shmemAddr*/)> perVectorCallback;
+  // Initialize variables in a safe scope. These variables will be bound
+  // to the lambda function.
+  // TODO: Enforce that crossGrain=True is only for AMD
+  unsigned int numElementsPerIter = 1;
+  unsigned int innerVectorization = 1;
+  unsigned int val_counter = 0;
   if (!crossGrain) {
-    perVectorCallback = [&](VectorType vecTy, Value vecAddr) {
+    perVectorCallback = [&srcVals, &elemLlvmTy, &loc, &rewriter](VectorType vecTy, Value vecAddr) {
```

Comment:
You don't need to make capture explicit for this to work, but it just felt safer.